### DISCUSSION TITLE
Small leak detection (e.g. seeping toilet)

### DISCUSSION THREAD

**Discussion by cbho** [jomjol/AI-on-the-edge-device#1320] [Ideas]
Not sure how this should be done, but:

What about making some kind of "Minor leak detection" function?
That is, if there is a very slow increase in the reading over a certain time, then a binary sensor could alert the user about possible "invisible" leaks?

I think there could be to different approaches:
1) Defining a certain volume and a certain time and raising the alert, if more than nothing, but less than the defined volume has been used during the defined time. That would however be prone to error since ordinary use of water can vary as much as it does.

2) Monitoring the rate and raise the alert, if the rate has been different from 0 and steady for a certain amount of time (that would actually allow for detecting leaks of any size).
Looking at the graph from my watermeter (example below) I've used app. 490 liters during the last two days (graph below).
That volume could have been used by a single toilet seeping - but the graph would have been like the red dotted line, not like the black illustrating varied everyday use.

![image](https://user-images.githubusercontent.com/29979509/201144544-651a706d-ad2b-4f75-bbd0-1651a67cb54c.png)

The challenge would of course be, to allow for a mix of those graphs, that is, to identify a constant rate usage (the leak) between the "ordinary use" parts of the graph.

As stated initially, I'm not sure how this sould be done (neither the math, nor the implementation), but am at the same time certain, that to somebody else, this is pretty trivial to make ;-)

Regards,
Chr.

**Comment by mad2xlc** [comment]
I am just thinking of an approach how I would solve this in Home Assistant. I don't know if you something similar.
What if you just check the consumption between midnight and 5 o'clock.
There the consumption should normally be at 0 liters. But if more is consumed, you can issue a warning.

**Reply by caco3** [reply]
I agree this could be a handy monitor.
I also have a (50 year old) toilet, which sometimes loses around 10 liters over night.

But I really think this should be implemented outside of the device for following reasons:
 - We don't have much application space left (use 97% already)
 - It adds another complexity
 - Is much easier to implement in eg. homeassistant and provides more flexibility.

**Comment by cbho** [comment]
@caco3: The space-issue is the most important aspect of this, I guess there will be other features/enhancements more worthy of getting a bite of this expensive bytes :-)

@mad2xlc: I get your point and think that could be one way to go. However, I have a water softener that regenerates when necessary, that is at rather random times (although preferably during the night) as well as weashing my laundry is often during the night (when power prices are low). Of course, adding some logic ind the equation that dismiss the calculation, if some of these processes have run during the time measured, would solve that problem. Still, I think it would be more elegant to figure out, how to measure the leak itself. Maybe it could be something by comparing delta-values every x minutes with the mean of those during a certain time and raise an alarm, if the delta value equals the mean of the delta values or something like that...

Anyway, will move on to the Home Assistant forum for further discussion/development on this topic.
Actually, I just found this discussion [Detect small leak from liters/min sensor](https://community.home-assistant.io/t/detect-small-leak-from-liters-min-sensor/386675) :-)
